apiserver/src/main/java/org/dependencytrack/resources/v1/AdvisoriesResource.java

```diff
@@ -0,0 +1,380 @@
+    public Response markAdvisoryAsSeen(@PathParam("advisoryId") String advisoryId) {
+        try (QueryManager qm = new QueryManager()) {
+            final Advisory entity = qm.getObjectById(Advisory.class, advisoryId);
+            if (entity != null) {
+                entity.setSeen(true);
+                qm.persist(entity);
+                return Response.ok(entity).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("The advisory could not be found.").build();
+            }
+        }
+    }
+
+    @GET
+    @Path("/{advisoryId}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Returns the details of an advisory", description = "<p>Requires permission <strong>VULNERABILITY_ANALYSIS_READ</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Details of given advisory"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
+    @PermissionRequired(Permissions.Constants.VULNERABILITY_ANALYSIS_READ)
+    public Response getAdvisoryById(
+            @Parameter(description = "The ID of the advisory to view", schema = @Schema(type = "string", format = "long"), required = true) @PathParam("advisoryId") String advisoryId) {
+        try (QueryManager qm = new QueryManager()) {
+            final var advisoryEntity = qm.getObjectById(Advisory.class, advisoryId);
+
+            if (advisoryEntity == null) {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("The requested advisory could not be found.")
+                        .build();
+            } else {
+                List<AdvisoryDao.ProjectRow> affectedProjects = withJdbiHandle(getAlpineRequest(),
+                        handle -> handle.attach(AdvisoryDao.class)
+                                .getProjectsByAdvisory(advisoryEntity.getId()));
+
+                List<AdvisoryDao.VulnerabilityRow> vulnerabilities = withJdbiHandle(getAlpineRequest(),
+                        handle -> handle.attach(AdvisoryDao.class)
+                                .getVulnerabilitiesByAdvisory(advisoryEntity.getId()));
+
+                Long numAffectedComponentsBoxed = withJdbiHandle(getAlpineRequest(),
+                        handle -> handle.attach(AdvisoryDao.class)
+                                .getAmountFindingsTotal(advisoryEntity.getId()));
+                long numAffectedComponents = numAffectedComponentsBoxed != null
+                        ? numAffectedComponentsBoxed
+                        : 0L;
+
+                return Response.ok(new AdvisoryDao.AdvisoryResult(
+                        advisoryEntity,
+                        affectedProjects,
+                        numAffectedComponents,
+                        vulnerabilities)).build();
+            }
+        }
```

Comment:
Use either the JDO API (`QueryManager`), *or* JDBI, but don't mix them in the same block. They cannot share transactions or connections. Keeping a handle of both of them open wastes resources.